Content.Shared/Whitelist/EntityWhitelistSystem.cs

```diff
@@ -49,10 +49,12 @@ public bool IsValid(EntityWhitelist list, EntityUid uid)
     {
         if (list.Components != null)
         {
-            var regs = StringsToRegs(list.Components);
-
-            list.Registrations ??= new List<ComponentRegistration>();
-            list.Registrations.AddRange(regs);
+            if (list.Registrations == null)
```

Comment:
I feel that this should make this a bit safer. Unfortunately, we can't efficiently track each array element value with this API.
```suggestion
            if (list.Registrations == null || list.Registrations.Count != list.Components.Length)
```